From c2215f0116b5ede334f3958c974340424a332cd9 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Wed, 30 Mar 2016 13:06:50 -0400
Subject: [PATCH 41/41] ks2: crypto: fix tx_dma_desc_cnt bug

The tx_dma_desc_cnt counter is to maintain number of free tx descriptors.
In case of error driver has to adjust the counter.
This commit fixes a bug of this adjustment.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 drivers/crypto/keystone-sa-utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/keystone-sa-utils.c b/drivers/crypto/keystone-sa-utils.c
index be4e532..d716621 100644
--- a/drivers/crypto/keystone-sa-utils.c
+++ b/drivers/crypto/keystone-sa-utils.c
@@ -1434,7 +1434,7 @@ err_1:
 	if (req_ctx)
 		kmem_cache_free(pdata->dma_req_ctx_cache, req_ctx);
 err_0:
-	atomic_add((sg_nents - SA_NUM_DMA_META_ELEMS), &pdata->tx_dma_desc_cnt);
+	atomic_add(sg_nents, &pdata->tx_dma_desc_cnt);
 	atomic_inc(&pdata->stats.tx_dropped);
 	return ret;
 }
-- 
1.9.1

